service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonCaseDataExtractionServiceTest.java

```diff
@@ -0,0 +1,82 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+
+import java.time.LocalDateTime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createHearingBooking;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {CommonCaseDataExtractionService.class, DateFormatterService.class})
+class CommonCaseDataExtractionServiceTest {
+
+    private final CommonCaseDataExtractionService commonCaseDataExtractionService;
+    private HearingBooking hearingBooking;
+
+    @Autowired
+    CommonCaseDataExtractionServiceTest(
+        CommonCaseDataExtractionService commonCaseDataExtractionService) {
+        this.commonCaseDataExtractionService = commonCaseDataExtractionService;
+    }
+
+    @Test
+    void shouldReturnTheFormattedDateWhenStartAndEndDateAreNotTheSame() {
+        hearingBooking = createHearingBookingWithTimesOnDifferentDays();
+
+        final String hearingDate = commonCaseDataExtractionService.getHearingDate(hearingBooking);
+
+        assertThat(hearingDate).isEmpty();
```

Comment:
as commented out somewhere else is confusing a bit. 🤔 maybe if we called it getHearingDateIfHearingsOnSameDay (hah a bit long, not sure really, have a thought about it, but not fan of the interaction)

Reply:
Yeah something like that sounds good, a lot of our method names get quite long so I wouldn't worry about that 😄

Reply:
Left the method name the same but made it return an `Optional<String>` as pointed out in a previous comment